proto/auth.proto

```diff
@@ -69,8 +69,37 @@ message _GenerateApiTokenRequest {
         Expires expires = 2;
     }
 
-    // session token recieved from `momento login` command
-    string session_token = 3;
+    // Aliases for categories of functionality.
+    enum Role {
+      PermitNone = 0;
```

Comment:
I'm unsure of the necessity for `permit none`

```
{
    permissions: {
      cache: {
          role: read
      }
}
```

To me means only cache read is allowed; no topics or control plane is allowed. I don't think you should have to do
```
{
   permissions: {
      cache: {
         role: read
       }
     topics: {
        role: permit none
     }
     control: {
        role: permit none
    }
  }
}
```

Because if we do this, then say we add a new family of apis called idk "orgs." When I originally created my token, my intention was to only have it be able to read from caches and do nothing else. But since `orgs` isn't in this list with a `permit none` then the token would be allowed to access these new APIs.

Reply:
I think the concern is that by default the enum parses to 0, so if we unmarshal it and it doesnt exist, then it would be set to `ReadWrite`. But if its unset is should be `NoPerms`

Reply:
oh right protobuf defaults. ok, I think that makes sense!

Reply:
oh man, this is a good point, if the default behavior is `permit none` shouldn't everything be named `permissions`?

`permissions`
`cachePermissions`
`topicPermissions`

the absence of a permission means deny

Reply:
its a whitelist, not a blacklist, which is where the word `restriction` is coming from

Reply:
this is why naming things "restrictions" confuses the hell out of me 😬

Reply:
if `permissions` is nonzero length, then the token is restricted.

The first layer is "permit" semantic, allowing access to cache/topic/whatever feature set.

The second layer is "restrict" semantic, narrowing access to the cache/topic/whatever feature set.

Reply:
lol both work, it's a yanny vs laurel situation:

- `restrictions` : this is the set of things users are "restricted to do", everything else is not allowed
- `permissions`: this is the set of things users are "permitted to do", everything else is not allowed

I don't care which one, I just think we should be consistent and not use both